File: alphabit-coding-challenge-03/walls/solutions/walls.py
n, m, ident, d, e, t, a, wid = 0, 0, 0, 0, 0, 0, 0, 0
wallarr = l = [None] * 100
wavearr = l = [None] * 100
flag = False

n, m = (int(x) for x in input().split(" "))
for i in range(n):
    ident, d, e= (int(x) for x in input().split(" "))
    tmp = [d, e]
    wallarr[ident - 1] = tmp
for i in range(m):
    t, a = (int(x) for x in input().split(" "))
    tmp = [t, a]
    wavearr[i] = tmp

ident = n - 1

a = wavearr[wid][1]

for i in range(n):
    wallarr[i][0] += wallarr[i][1] * wavearr[0][0]

while ((wid < m) and (not(flag))):
    if wallarr[ident][0] >= a:
        for i in range(n):
            wallarr[i][0] += wallarr[i][1]
        wid += 1
        if wid < m:
            a += wavearr[wid][1]
    else:
        if ident == 0:
            flag = True
        else:
            ident -= 1
            for  i in range(n):
                wallarr[i][0] += wallarr[i][1]
if flag:
    print("humanity in danger ")
else:
    print(ident + 1)

File: alphabit-coding-challenge-03/walls/solutions/walls.c
#include <stdio.h>
#include <stdlib.h>




int main()
{

    int n,m,id,d,e,t,a,wid;
    int wallarr[101][2];
    int wavearr [101][2];
    int flag=0;

    // read the input
    scanf("%d %d",&n,&m);
    for(int i =0;i<n;i++){
        scanf("%d %d %d",&id,&d,&e);
        wallarr[id-1][0]= d;
        wallarr[id-1][1]= e;
        };
    for(int i =0;i<m;i++){
    scanf("%d %d",&t,&a);
    wavearr[i][0]= t;
    wavearr[i][1]= a;
    };
    // traitment
    id = n-1;
    wid = 0;
    a=wavearr[wid][1];

     for (int i =0;i<n;i++){
        wallarr[i][0] += wallarr[i][1]*wavearr[0][0];
           }
    while ((wid<m)&&(!flag)){
            if (wallarr[id][0]>=a){
                for (int i =0;i<n;i++){
                wallarr[i][0] += wallarr[i][1];
                }
                wid++;
                if(wid<m){a+=wavearr[wid][1];}



            }else{
                if (id == 0){flag = 1;}else{
                     id--;
                     for (int i =0;i<n;i++){
                    wallarr[i][0] += wallarr[i][1];
                    }
                }

            }
    }




// write the output

if (flag){printf("humanity in danger ");}else{
    printf("%d",id+1);
}

    return 0;
}


